### ISSUE TITLE
Snapshot to S3 uploads files but shows incomplete

### ISSUE THREAD

**Issue by MikaelSmith** [replicatedhq/kots#942]
I attempted to do a backup to S3 (using a kURL cluster with Velero setup)
![Screen Shot 2020-08-11 at 2 54 24 PM](https://user-images.githubusercontent.com/8162496/90074556-5b292780-dcb0-11ea-9ce3-8eed45491ec1.png)

The files appear to be uploaded
![Screen Shot 2020-08-11 at 2 53 53 PM](https://user-images.githubusercontent.com/8162496/90074529-51072900-dcb0-11ea-9120-d07b48202bbc.png)

but the Snapshot says it's incomplete
![Screen Shot 2020-08-11 at 2 56 15 PM](https://user-images.githubusercontent.com/8162496/90074598-6c723400-dcb0-11ea-9bac-ca5c342deeae.png)

Restic logs seem to report that they're looking in the wrong place for some config
```
time="2020-08-11T21:42:12Z" level=info msg="Setting log-level to INFO"
time="2020-08-11T21:42:12Z" level=info msg="Starting Velero restic server v1.2.0 (5d008491bbf681658d3e372da1a9d3a21ca4c03c)" logSource="pkg/cmd/cli/restic/server.go:62"
time="2020-08-11T21:42:12Z" level=info msg="Starting controllers" logSource="pkg/cmd/cli/restic/server.go:156"
time="2020-08-11T21:42:12Z" level=info msg="Controllers started successfully" logSource="pkg/cmd/cli/restic/server.go:199"
time="2020-08-11T21:42:12Z" level=info msg="Starting controller" controller=pod-volume-backup logSource="pkg/controller/generic_controller.go:76"
time="2020-08-11T21:42:12Z" level=info msg="Waiting for caches to sync" controller=pod-volume-backup logSource="pkg/controller/generic_controller.go:79"
time="2020-08-11T21:42:12Z" level=info msg="Starting controller" controller=pod-volume-restore logSource="pkg/controller/generic_controller.go:76"
time="2020-08-11T21:42:12Z" level=info msg="Waiting for caches to sync" controller=pod-volume-restore logSource="pkg/controller/generic_controller.go:79"
time="2020-08-11T21:42:12Z" level=info msg="Caches are synced" controller=pod-volume-backup logSource="pkg/controller/generic_controller.go:83"
time="2020-08-11T21:42:12Z" level=info msg="Caches are synced" controller=pod-volume-restore logSource="pkg/controller/generic_controller.go:83"
time="2020-08-11T21:42:14Z" level=info msg="Backup starting" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:178" name=cd4pe-fw5qn-6fsqh namespace=velero
time="2020-08-11T21:42:14Z" level=info msg="Looking for most recent completed pod volume backup for this PVC" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:301" name=cd4pe-fw5qn-6fsqh namespace=velero pvcUID=4017f8aa-7422-4804-9a41-93aa4e14fff3
time="2020-08-11T21:42:14Z" level=info msg="Found most recent completed pod volume backup for PVC" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:330" name=cd4pe-fw5qn-6fsqh namespace=velero parentPodVolumeBackup=cd4pe-d8zq6-ndwpw parentSnapshotID=a75186ac pvcUID=4017f8aa-7422-4804-9a41-93aa4e14fff3
time="2020-08-11T21:42:14Z" level=info msg="Setting --parent flag for this backup" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:249" name=cd4pe-fw5qn-6fsqh namespace=velero parentSnapshotID=a75186ac
time="2020-08-11T21:42:14Z" level=error msg="Error running command=restic backup --repo=s3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe --password-file=/tmp/velero-restic-credentials-cd4pe489012768 --cache-dir=/scratch/.cache/restic . --tag=pod=minio-0 --tag=pod-uid=ad5dd4f5-dd34-459e-acc7-5e520d63986e --tag=pvc-uid=4017f8aa-7422-4804-9a41-93aa4e14fff3 --tag=volume=data --tag=backup=cd4pe-fw5qn --tag=backup-uid=9c1b6af0-650a-40b3-8d3d-275b1e098ba1 --tag=ns=cd4pe --host=velero --json --parent=a75186ac, stdout=, stderr=Fatal: unable to open config file: Stat: The access key ID you provided does not exist in our records.\nIs there a repository at the following location?\ns3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe\n" backup=velero/cd4pe-fw5qn controller=pod-volume-backup error="unable to find summary in restic backup command output" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/exec_commands.go:169" error.function=github.com/vmware-tanzu/velero/pkg/restic.getSummaryLine logSource="pkg/controller/pod_volume_backup_controller.go:261" name=cd4pe-fw5qn-6fsqh namespace=velero
time="2020-08-11T21:42:14Z" level=info msg="Backup starting" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:178" name=cd4pe-fw5qn-r7ftp namespace=velero
time="2020-08-11T21:42:14Z" level=info msg="Looking for most recent completed pod volume backup for this PVC" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:301" name=cd4pe-fw5qn-r7ftp namespace=velero pvcUID=c10d230c-dda3-44d6-ba54-3b74f5d6cc39
time="2020-08-11T21:42:14Z" level=info msg="Found most recent completed pod volume backup for PVC" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:330" name=cd4pe-fw5qn-r7ftp namespace=velero parentPodVolumeBackup=cd4pe-d8zq6-574w2 parentSnapshotID=5619dfd3 pvcUID=c10d230c-dda3-44d6-ba54-3b74f5d6cc39
time="2020-08-11T21:42:14Z" level=info msg="Setting --parent flag for this backup" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:249" name=cd4pe-fw5qn-r7ftp namespace=velero parentSnapshotID=5619dfd3
time="2020-08-11T21:42:14Z" level=error msg="Error running command=restic backup --repo=s3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe --password-file=/tmp/velero-restic-credentials-cd4pe753590015 --cache-dir=/scratch/.cache/restic . --tag=pod=minio-1 --tag=pod-uid=040a589e-531f-4317-b387-1aba45acb7f2 --tag=pvc-uid=c10d230c-dda3-44d6-ba54-3b74f5d6cc39 --tag=volume=data --tag=backup=cd4pe-fw5qn --tag=backup-uid=9c1b6af0-650a-40b3-8d3d-275b1e098ba1 --tag=ns=cd4pe --host=velero --json --parent=5619dfd3, stdout=, stderr=Fatal: unable to open config file: Stat: The access key ID you provided does not exist in our records.\nIs there a repository at the following location?\ns3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe\n" backup=velero/cd4pe-fw5qn controller=pod-volume-backup error="unable to find summary in restic backup command output" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/exec_commands.go:169" error.function=github.com/vmware-tanzu/velero/pkg/restic.getSummaryLine logSource="pkg/controller/pod_volume_backup_controller.go:261" name=cd4pe-fw5qn-r7ftp namespace=velero
time="2020-08-11T21:42:14Z" level=info msg="Backup starting" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:178" name=cd4pe-fw5qn-4kv6v namespace=velero
time="2020-08-11T21:42:14Z" level=info msg="Looking for most recent completed pod volume backup for this PVC" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:301" name=cd4pe-fw5qn-4kv6v namespace=velero pvcUID=1bf93c56-9733-4767-8565-0e5d337d0480
time="2020-08-11T21:42:14Z" level=info msg="Found most recent completed pod volume backup for PVC" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:330" name=cd4pe-fw5qn-4kv6v namespace=velero parentPodVolumeBackup=cd4pe-d8zq6-ls857 parentSnapshotID=b6348db2 pvcUID=1bf93c56-9733-4767-8565-0e5d337d0480
time="2020-08-11T21:42:14Z" level=info msg="Setting --parent flag for this backup" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:249" name=cd4pe-fw5qn-4kv6v namespace=velero parentSnapshotID=b6348db2
time="2020-08-11T21:42:14Z" level=error msg="Error running command=restic backup --repo=s3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe --password-file=/tmp/velero-restic-credentials-cd4pe759988818 --cache-dir=/scratch/.cache/restic . --tag=pod-uid=27769302-99fe-4c61-a6b8-0b3fc05c57c1 --tag=pvc-uid=1bf93c56-9733-4767-8565-0e5d337d0480 --tag=volume=data --tag=backup=cd4pe-fw5qn --tag=backup-uid=9c1b6af0-650a-40b3-8d3d-275b1e098ba1 --tag=ns=cd4pe --tag=pod=minio-2 --host=velero --json --parent=b6348db2, stdout=, stderr=Fatal: unable to open config file: Stat: The access key ID you provided does not exist in our records.\nIs there a repository at the following location?\ns3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe\n" backup=velero/cd4pe-fw5qn controller=pod-volume-backup error="unable to find summary in restic backup command output" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/exec_commands.go:169" error.function=github.com/vmware-tanzu/velero/pkg/restic.getSummaryLine logSource="pkg/controller/pod_volume_backup_controller.go:261" name=cd4pe-fw5qn-4kv6v namespace=velero
time="2020-08-11T21:42:14Z" level=info msg="Backup starting" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:178" name=cd4pe-fw5qn-gsxcf namespace=velero
time="2020-08-11T21:42:14Z" level=info msg="Looking for most recent completed pod volume backup for this PVC" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:301" name=cd4pe-fw5qn-gsxcf namespace=velero pvcUID=5d019319-cb18-473e-bcd3-6464f7f50adf
time="2020-08-11T21:42:14Z" level=info msg="Found most recent completed pod volume backup for PVC" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:330" name=cd4pe-fw5qn-gsxcf namespace=velero parentPodVolumeBackup=cd4pe-d8zq6-fqngs parentSnapshotID=1decd397 pvcUID=5d019319-cb18-473e-bcd3-6464f7f50adf
time="2020-08-11T21:42:14Z" level=info msg="Setting --parent flag for this backup" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:249" name=cd4pe-fw5qn-gsxcf namespace=velero parentSnapshotID=1decd397
time="2020-08-11T21:42:14Z" level=error msg="Error running command=restic backup --repo=s3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe --password-file=/tmp/velero-restic-credentials-cd4pe122225545 --cache-dir=/scratch/.cache/restic . --tag=pod-uid=0a72d169-7023-44b7-880a-5852b6de18f7 --tag=pvc-uid=5d019319-cb18-473e-bcd3-6464f7f50adf --tag=volume=data --tag=backup=cd4pe-fw5qn --tag=backup-uid=9c1b6af0-650a-40b3-8d3d-275b1e098ba1 --tag=ns=cd4pe --tag=pod=minio-3 --host=velero --json --parent=1decd397, stdout=, stderr=Fatal: unable to open config file: Stat: The access key ID you provided does not exist in our records.\nIs there a repository at the following location?\ns3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe\n" backup=velero/cd4pe-fw5qn controller=pod-volume-backup error="unable to find summary in restic backup command output" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/exec_commands.go:169" error.function=github.com/vmware-tanzu/velero/pkg/restic.getSummaryLine logSource="pkg/controller/pod_volume_backup_controller.go:261" name=cd4pe-fw5qn-gsxcf namespace=velero
time="2020-08-11T21:42:15Z" level=info msg="Backup starting" backup=velero/cd4pe-fw5qn controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:178" name=cd4pe-fw5qn-tw8jv namespace=velero
time="2020-08-11T21:42:15Z" level=error msg="Error running command=restic backup --repo=s3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe --password-file=/tmp/velero-restic-credentials-cd4pe183351124 --cache-dir=/scratch/.cache/restic . --tag=volume=backup --tag=backup=cd4pe-fw5qn --tag=backup-uid=9c1b6af0-650a-40b3-8d3d-275b1e098ba1 --tag=ns=cd4pe --tag=pod=postgres-0 --tag=pod-uid=a75e44c8-fbc1-4180-9a9e-222a03009477 --host=velero --json, stdout=, stderr=Fatal: unable to open config file: Stat: The access key ID you provided does not exist in our records.\nIs there a repository at the following location?\ns3:http://rook-ceph-rgw-rook-ceph-store.rook-ceph/velero/restic/cd4pe\n" backup=velero/cd4pe-fw5qn controller=pod-volume-backup error="unable to find summary in restic backup command output" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/exec_commands.go:169" error.function=github.com/vmware-tanzu/velero/pkg/restic.getSummaryLine logSource="pkg/controller/pod_volume_backup_controller.go:261" name=cd4pe-fw5qn-tw8jv namespace=velero
```

**Comment by divolgin** [comment]
@MikaelSmith how did you configure Velero?  Was it just through the UI: make changes, save, click Start Snapshot?

**Comment by ccaum** [comment]
It was set up through the GUI. Go to the Snapshots tab, click Settings, select the S3 option from the destination dropdown, and fill out the AWS creds.

Once that was done, a manual snapshot was started.

**Comment by divolgin** [comment]
ok, we'll try to reproduce it.

**Comment by salahalsaleh** [comment]
Resolved in KOTS v1.19.0
